servo/connectors/kube_metrics.py

```diff
@@ -407,80 +407,76 @@ async def periodic_measure(
                     cpu_resources = target_resource_container.get_resource_requirements(
                         "cpu"
                     )
+                    # Set requests = limits if not specified
+                    cpu_request = cpu_resources[ResourceRequirement.request]
+                    if cpu_request is None:
```

Comment:
Using := is more pythonic btw, you just have to use parenthesis to ensure the order of operations doesn't wind up assigning the boolean expression to your variable instead of the value you're testing
```suggestion
                    if (cpu_request := cpu_resources[ResourceRequirement.request]) is None:
```